core/src/main/java/com/onelogin/saml2/authn/SamlResponse.java

```diff
@@ -241,6 +241,15 @@ public boolean isValid(String requestId) {
 								+ ", does not match the ID of the AuthNRequest sent by the SP: " + requestId, ValidationError.WRONG_INRESPONSETO);
 				}
 
+				// Check SAML version on the response
+				NodeList assertions = queryAssertion("");
```

Comment:
We can move the 2.0 version check on assertion after the validateNumAssertions, so we only need to check the first assertion found

Reply:
The 2.0 version check is already performed after the `validateNumAssertions()` call (which is performed at line 206), so if there are more than one assertions, the validation will break before that and no version check will be made. Nevertheless, I'm iterating here for consistency and safety, but in practice the loop will always visit just one node, the first and only one. Let me know if I should rather change the check just over `assertions.item(0)` unceremoniously...

Reply:
I think we can:
```
// Check SAML version on the assertion
NodeList assertions = queryAssertion("");
Node versionAttribute = assertions.item(0).getAttributes().getNamedItem("Version");
if (versionAttribute == null || !"2.0".equals(versionAttribute.getNodeValue())) {
    throw new ValidationError("Unsupported SAML Version on Assertion.", ValidationError.UNSUPPORTED_SAML_VERSION);
}
```

reducing code... at the end, this element has no risk at all.

Reply:
Done.